src/main/kotlin/no/nav/familie/ef/sak/vilkår/dto/TidligereVedtaksperioderDto.kt

```diff
@@ -33,4 +47,65 @@ fun TidligereInnvilgetVedtak.tilDto() =
         harTidligereOvergangsstønad = this.harTidligereOvergangsstønad,
         harTidligereBarnetilsyn = this.harTidligereBarnetilsyn,
         harTidligereSkolepenger = this.harTidligereSkolepenger,
+        periodeHistorikkOvergangsstønad = this.periodeHistorikkOvergangsstønad.tilDto(),
     )
+
+private fun List<GrunnlagsdataPeriodeHistorikk>.tilDto() = this.map { it.tilDto() }
+    .slåSammenPåfølgendePerioderMedLikPeriodetype()
+    .sortedByDescending { it.fom }
+
+private fun GrunnlagsdataPeriodeHistorikk.tilDto() = GrunnlagsdataPeriodeHistorikkDto(
+    periodeType = this.periodeType.name,
+    fom = this.fom,
+    tom = this.tom,
+    antMnd = mndMedBeløp(periodeType, beløp, fom, tom),
+    antallMndUtenBeløp = mndUtenBeløp(periodeType, beløp, fom, tom),
+)
+
+private fun mndUtenBeløp(
+    periodeType: VedtaksperiodeType,
+    beløp: Int,
+    fom: LocalDate,
+    tom: LocalDate,
+) = when (beløp == 0 && !periodeType.midlertidigOpphørEllerSanksjon()) {
+    true -> Månedsperiode(fom, tom).lengdeIHeleMåneder()
+    false -> 0
+}
+
+private fun mndMedBeløp(
+    periodeType: VedtaksperiodeType,
+    beløp: Int,
+    fom: LocalDate,
+    tom: LocalDate,
+) = when (beløp != 0 || periodeType.midlertidigOpphørEllerSanksjon()) {
+    true -> Månedsperiode(fom, tom).lengdeIHeleMåneder()
+    false -> 0
+}
+
+fun List<GrunnlagsdataPeriodeHistorikkDto>.slåSammenPåfølgendePerioderMedLikPeriodetype(): List<GrunnlagsdataPeriodeHistorikkDto> {
```

Comment:
Ser bra og riktig ut, men noe tung å lese. Ville det vært mulig å løse, og lettere å lese, med zipWithNext i stedet for fold?
Selvfølgelig meget valgfritt om det er noe som er verdt å bruke tid på. Fagtorsdag utfordring? ;D